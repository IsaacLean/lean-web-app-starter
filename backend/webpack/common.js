/**
 * This is the webpack configuration that is used for all modes.
 */
const path = require('path')
const setupNodeExternals = require('ljas-webpack/setupNodeExternals')
const { compileTS } = require('ljas-webpack')
const { merge } = require('webpack-merge')

module.exports = merge([
  {
    entry: './src/main.ts',
  },

  compileTS({ include: path.resolve(__dirname, '../src') }),

  setupNodeExternals(),
])
