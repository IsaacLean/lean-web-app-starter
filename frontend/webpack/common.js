/**
 * This is the webpack configuration that is used for all modes.
 */
const buildHTML = require('ljas-webpack/buildHTML')
const path = require('path')
const { compileReact } = require('ljas-webpack')
const { merge } = require('webpack-merge')

module.exports = merge([
  {
    entry: './src/main.tsx',
  },

  buildHTML('Lean JavaScript Application Starter', 'src/index.html'),

  compileReact(path.resolve(__dirname, '../src')),
])
