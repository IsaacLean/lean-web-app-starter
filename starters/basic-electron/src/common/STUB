STUB: All code shared across main, preload, and renderer directories goes in here.
