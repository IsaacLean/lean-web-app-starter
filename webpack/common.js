/**
 * This is the webpack configuration that is used for all modes.
 */
const { merge } = require('webpack-merge')
const buildHTML = require('../ljas-webpack/buildHTML')

module.exports = merge([
  {
    entry: './src/main.tsx',
  },

  buildHTML(),
])
