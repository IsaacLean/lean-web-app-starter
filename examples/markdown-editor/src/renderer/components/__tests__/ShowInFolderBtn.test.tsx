import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { MOCK_FOOBAR_FILE_PATH } from '../../../common/MOCK_DATA'
import ShowInFolderBtn from '../ShowInFolderBtn'

beforeEach(() => {
    // Mock the API
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    window.api = {
        showInFolder: jest.fn(),
    }
})

/**
 * Setup user-event and render the component.
 */
const setupTest = (filePath?: string) => {
    const user = userEvent.setup()
    const { asFragment } = render(<ShowInFolderBtn filePath={filePath} />)

    return { asFragment, user }
}

test('ShowInFolderBtn component matches snapshot', () => {
    const { asFragment } = setupTest()
    expect(asFragment()).toMatchSnapshot()
})

test('ShowInFolderBtn component does not open a folder when clicked while disabled', async () => {
    // Render without filePath prop to simulate that file is not opened
    const { user } = setupTest()

    // Expect showInFolder to start with 0 calls
    expect(
        (
            window.api.showInFolder as jest.MockedFunction<
                typeof window.api.showInFolder
            >
        ).mock.calls,
    ).toHaveLength(0)

    await user.click(screen.getByRole('button'))

    // Expect showInFolder to remain with 0 calls as the click should have been on a disabled button
    expect(
        (
            window.api.showInFolder as jest.MockedFunction<
                typeof window.api.showInFolder
            >
        ).mock.calls,
    ).toHaveLength(0)
})

test('ShowInFolderBtn component opens the folder the currently opened file is located in when clicked while enabled', async () => {
    // Render with filePath prop to simulate that file is opened
    const { user } = setupTest(MOCK_FOOBAR_FILE_PATH)

    // Expect showInFolder to start with 0 calls
    expect(
        (
            window.api.showInFolder as jest.MockedFunction<
                typeof window.api.showInFolder
            >
        ).mock.calls,
    ).toHaveLength(0)

    await user.click(screen.getByRole('button'))

    // Expect showInFolder to now have 1 call
    expect(
        (
            window.api.showInFolder as jest.MockedFunction<
                typeof window.api.showInFolder
            >
        ).mock.calls,
    ).toHaveLength(1)
})
