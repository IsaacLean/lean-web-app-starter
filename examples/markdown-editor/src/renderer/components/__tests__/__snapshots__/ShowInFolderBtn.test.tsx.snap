// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowInFolderBtn component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn"
    disabled=""
  >
    Show In Folder
  </button>
</DocumentFragment>
`;
